### PR TITLE
Adding improved user analytics dashboard, Fixing https://wso2.org/jira/browse/PC-155

### PR DESCRIPTION
**Pull request by dewmini**

### CODE REVIEW

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/index.xml [hunk 1]
```diff
@@ -60,11 +62,14 @@
 </head>
 
 <body>
-    <div class="panel panel-default">
-        <div class="panel-body">
+    <div>
+        <div class="panel-body" style="width:100%; height:95%">
```
Comment (prasa7) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/index.xml]: Remove these inline styles and add these to a CSS class.

File: modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/*a:hover{*/
+    /*text-decoration: none;*/
+/*}*/
+
+@media (min-width: 400px) {
+    .form-inline .form-control {
+        display: inline-block;
+        width: auto;
+        vertical-align: middle;  }
```
Comment (prasa7) [inline modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css]: formatting issue. "}" should be in the below line

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_task_id/index.xml [hunk 3]
```diff
@@ -87,7 +92,7 @@
                     <button class="btn btn-primary" onclick="drawGraph()">Apply Filter</button>
                 </div>
             </div>
-            <div id="chartA"></div>
+            <div id="chartA" style="width:100%; height:100%; padding-top:20px"></div>
```
Comment (prasa7) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_task_id/index.xml]: Remove inline styles.

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/index.xml (hunk 1 above)
Comment (dewmini) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/index.xml]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css (hunk 2 above)
Comment (dewmini) [inline modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_task_id/index.xml (hunk 3 above)
Comment (dewmini) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_task_id/index.xml]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+.panel-body a {
+    color: rgb(52, 59, 89);
+    text-decoration: none;
+}
+
+.fw-filter:before {
+    content: '\e634';
+}
+
+/*a:hover{*/
```
Comment (dakshika) [inline modules/extensions/apps/portal/libs/pc-analytics-libs/css/custom.css]: remove commented css

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/js/actions.js [hunk 5]
```diff
@@ -47,14 +47,18 @@ function drawGraph() {
                     var jsonArrObj = JSON.parse('[' + responseStr + ']');
                     jsonObj[0].data = jsonArrObj;
 
+                    config.width = $('#chartA').width();
```
Comment (dakshika) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/avg_execution_time_vs_task_id/js/actions.js]: cache the selectors

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml [hunk 6]
```diff
@@ -90,21 +85,14 @@
 </head>
 
 <body>
-    <div class="panel panel-default">
-        <div class="panel-body">
+    <div>
+        <div class="panel-body" style="width:100%; height:95%">
```
Comment (dakshika) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml]: avoid inline styles

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml [hunk 7]
```diff
@@ -125,28 +113,22 @@
...
-        var callbackmethod = function(event, item) {
-            alert('chart clicked');
+            maxLength: 200,
+            xAxisAngle:true,
+            padding: {"top": 10, "left": 80, "bottom": 150, "right": 0},
+            transform:[60,70]
         }
 
         window.onload = function() {
+            $('#collapse').collapse("hide");
```
Comment (dakshika) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml]: don't use common ID when you are creating IDs

File: modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml (hunk 6 above)
Comment (dewmini) [inline modules/extensions/apps/portal/store/carbon.super/fs/gadget/task_instance_count_vs_date/index.xml]: This is fixed by https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

**Pull request merged**